### DISCUSSION TITLE
谁知道 gpu 和分部署是靠 golang哪个包实现的？

### DISCUSSION THREAD

**Discussion by icetech233** [milvus-io/milvus#36232] [Q&A and General discussion]
谁知道 gpu 和分部署是靠 golang哪个包实现的？

**Comment by yhmo** [comment]
Milvus 的上层调度主要用Go，下层向量计算向量索引主要用C++。
下层主要依赖一个单独的库：knowhere，和GPU相关的代码可以翻这里：
https://github.com/zilliztech/knowhere/tree/main/src/index/gpu
https://github.com/zilliztech/knowhere/tree/main/src/index/gpu_raft
